devel/validate-platforms

```diff
@@ -40,20 +40,41 @@ main() {
         echo "[$platform] Generating report file: $report"
         mkdir -p "$(dirname "$report")"
 
+        # Versions to track from `pip list`
+        tracked_packages=(bcbio-gff
+            biopython
+            cycler
+            evofr
+            nextstrain-augur
+            nextstrain-cli
+            numpy
+            numpyro
+            pandas
+            phylo-treetime
+            pyfastx
+            pysam
+            rethinkdb
+            scikit-learn
+            scipy
+            snakemake
+            xopen
+        )
+        tracked_packages_string=${tracked_packages[*]}
+        tracked_packages_delim=${tracked_packages_string// /|}
+
         # Create a report file for the platform.
         echo "[$platform] Determining software versions..."
         docker-run "$platform" bash -c '
             PS4="\$ "
             set -x
 
-            nextstrain --version
             nextalign --version
             nextclade --version
-            augur --version
             auspice --version
-            python3 -c "from importlib.metadata import version; print(version(\"evofr\"))"
-            datasets --version
-            dataformat version
+            # datasets --version
+            # dataformat version
+
+            pip list 2>&1 | grep -E "'"$tracked_packages_delim"'"
```

Comment:
```suggestion
            pip freeze 2>&1 | grep -E "'"$tracked_packages_delim"'"
```

I'd use `pip freeze` over `pip list` since it's a bit more verbose. See the difference in outputs:

`pip list`:

```diff
--- linux/amd64	2023-03-07 10:06:11.000000000 -0800
+++ linux/arm64	2023-03-07 10:06:15.000000000 -0800
@@ -9,6 +9,6 @@
 bcbio-gff                0.6.9
-biopython                1.81
+biopython                1.80
 cycler                   0.11.0
 evofr                    0.1.16
-nextstrain-augur         21.0.0      /nextstrain/augur
+nextstrain-augur         21.0.1      /nextstrain/augur
 nextstrain-cli           6.1.0.post1
```

`pip freeze`:

```diff
--- linux/amd64	2023-03-07 10:18:41.000000000 -0800
+++ linux/arm64	2023-03-07 10:18:43.000000000 -0800
@@ -9,6 +9,6 @@
 bcbio-gff==0.6.9
-biopython==1.81
+biopython==1.80
 cycler==0.11.0
 evofr==0.1.16
-# Editable install with no version control (nextstrain-augur==21.0.0)
+# Editable install with no version control (nextstrain-augur==21.0.1)
 nextstrain-cli==6.1.0.post1
```

Reply:
I'm not going to argue over output but to me this looks like a reformatting without any extra information at all - so not like `-vv` adding `info` and `debug` levels, but just adding extra words to `warn` and `error` 🙃 

It seems to be a 1:1 mapping but with less readable formatting